src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/SolutionExplorerInProcess.cs

```diff
@@ -553,21 +554,47 @@ private async Task<string> WaitForBuildToFinishAsync(IVsOutputWindowPane buildOu
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            await KnownUIContexts.SolutionExistsAndNotBuildingAndNotDebuggingContext;
+            var buildManager = await GetRequiredGlobalServiceAsync<SVsSolutionBuildManager, IVsSolutionBuildManager2>(cancellationToken);
+            using var semaphore = new SemaphoreSlim(1);
+            using var solutionEvents = new UpdateSolutionEvents(buildManager);
+            await semaphore.WaitAsync();
+            void HandleUpdateSolutionDone(bool succeeded, bool modified, bool canceled) => semaphore.Release();
+            solutionEvents.OnUpdateSolutionDone += HandleUpdateSolutionDone;
+            try
+            {
+                await semaphore.WaitAsync();
+            }
+            finally
+            {
+                solutionEvents.OnUpdateSolutionDone -= HandleUpdateSolutionDone;
+            }
 
             // Force the error list to update
             ErrorHandler.ThrowOnFailure(buildOutputWindowPane.FlushToTaskList());
 
-            var textView = (IVsTextView)buildOutputWindowPane;
+            if (buildOutputWindowPane is not IVsTextView textView)
```

Comment:
Emm, so why the type cast failed in the original test?

Reply:
The issue was that the build hadn't completed, so the output window hadn't activated.

Reply:
That's .. pretty weird. I probably need to look into the native code to understand how they do the  switching

Reply:
Emm still feels puzzled, it seems like COutputWindowPane is the underlying type for IVsOutputWindowPane, and it's OutputWindow that cound be casted to IVsTestView.

Reply:
Probably need to wait and see the result to correct me :)

Reply:
Had we invoked `pane.Activate()` ourselves then the cast would have succeeded. The test would still fail because the text would be empty because the build had not been performed.

Reply:
OK, I get my answer after reading the painful C++ code underhood.
Yeah each pane has a pointer points to the text view in it, later the cast is done by using QueryInterface method.
If the pane is not activated/initiaized we can't not do that cast with nullPtr